[docs] Added per page layout example with TypeScript (#27488)
vercel/next.js: docs/basic-features/layouts.md

> **Note**: This process is called [reconciliation](https://reactjs.org/docs/reconciliation.html), which is how React understands which elements have changed.

### With TypeScript

> When using TypeScript you must first create a new type for your pages which includes a `getLayout` function. Then you must create a new type for your `AppProps` which overrides the `Component` property to be of the previously created type.

Reviewer: ```suggestion
When using TypeScript, you must first create a new type for your pages which includes a `getLayout` function. Then, you must create a new type for your `AppProps` which overrides the `Component` property to use the previously created type.
```
